lnd_services.go

```diff
@@ -533,6 +565,60 @@ func (s *GrpcLndServices) waitForChainSync(ctx context.Context,
 	return <-update
 }
 
+// waitForChainNotifier blocks until the ChainNotifier RPC accepts block epoch
+// subscriptions and delivers at least one block height.
+func (s *GrpcLndServices) waitForChainNotifier(ctx context.Context,
+	timeout, pollInterval time.Duration) error {
+
+	mainCtx := ctx
+	if mainCtx == nil {
+		mainCtx = context.Background()
+	}
+
+	register := s.ChainNotifier.RegisterBlockEpochNtfn
+
+	for {
+		// Make new RegisterBlockEpochNtfn call.
+		subCtx, cancel := context.WithTimeout(mainCtx, timeout)
+		blockChan, errChan, err := register(subCtx)
+		if err != nil {
+			cancel()
+
+			return fmt.Errorf("register block epoch ntfn: %w", err)
+		}
+
+		// Wait for block height notification, which indicates success.
+		select {
+		case <-mainCtx.Done():
+			cancel()
```

Comment:
can be removed when we. use defer cancel() above

Reply:
Removed the call, used `defer` statement in `attempt` lambda.